includes/wikia/api/RevisionApiController.class.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+class RevisionApiController extends WikiaApiController {
+	const ROWS_PATTERN = '/<tr>(.*?)<\/tr>/s';
+	const CELLS_PATTERN = '/<td .*?class="(.*?)">(.*?)<\/td>/s';
+
+	/**
+	 * Gets diff between two revisions.
+	 * Also includes data about revision and article
+	 *
+	 * @param int $oldId old revision id (required)
+	 * @param int $newId new revision id (required)
+	 * @param bool $avatar should get user avatar
+	 * @param bool $oldRev should get data about old revision
+	 *
+	 * @throws MissingParameterApiException
+	 */
+	public function getRevisionsDiff() {
+		$oldId = $this->getRequest()->getVal( 'oldId', null );
+		$newId = $this->getRequest()->getVal( 'newId', null );
+
+		if ( empty( $oldId ) ) {
+			throw new MissingParameterApiException( 'oldId' );
+		}
+
+		if ( empty( $newId ) ) {
+			throw new MissingParameterApiException( 'newId' );
+		}
+
+		$avatar = $this->getRequest()->getVal( 'avatar', false );
+		$oldRev = $this->getRequest()->getVal( 'oldRev', false );
```

Comment:
I'd put `$this->getRequest()` in a local var, and also this one could be `$request->getBool( 'oldRev' );` since it's a boolean param, right?